policybot/pkg/gh/fromhook.go

```diff
@@ -189,3 +189,16 @@ func PullRequestReviewFromHook(prrp *hook.PullRequestReviewPayload) (*storage.Pu
 		State:               prrp.Review.State,
 	}, discoveredUsers
 }
+
+// Maps from a GitHub webhook event to a storage Test Flake. Also returns the set of
+// users discovered in the event in a map of {UserID:Login}.
+func TestResultFromHook(crp *hook.CheckRunPayload) (*storage.TestResult, map[string]string) {
```

Comment:
I think this function should go directly into resultTester.go because it is not in fact returning a fully initialized TestResult, it only fills in a few fields. The other functions in this package are all full conversions.

As a general rule within the code, the storage.* are considered immutable once created. This function is breaking that abstraction, as it returns a partially initialized struct.